server/routes/api.js

```diff
@@ -0,0 +1,73 @@
+const express = require('express')
+const router = express.Router()
+const Expense = require('../models/Expense.js')
+const moment = require('moment')
+
+
+router.get('/expenses/:d1?/:d2?', function (req, res) {
+    const d1 = req.query.d1
+    const d2 = req.query.d2
```

Comment:
Here it will be a good idea to destructuring. see [https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment](url)